### PR TITLE
avm2: Implement XMLList.parent

### PR DESCRIPTION
**Pull request by sleepycatcoding**

### CODE REVIEW

File: core/src/avm2/globals/XMLList.as [hunk 1]
```diff
@@ -22,6 +22,7 @@ package {
         AS3 native function toXMLString():String;
         AS3 native function toString():String;
         AS3 native function comments():XMLList;
+        AS3 native function parent():Object;
```
Comment (Lord-McSweeney) [inline core/src/avm2/globals/XMLList.as:25]: I'm pretty sure this is also defined as a method on the `prototype`

Comment (Lord-McSweeney) [reply core/src/avm2/globals/XMLList.as:25]: Also, the return type should be `*`, to allow for returning `undefined` (also, that's the signature in playerglobals)

Comment (sleepycatcoding) [reply core/src/avm2/globals/XMLList.as]: Fixed.

**Review by Lord-McSweeney — APPROVED** [review]

**Review by Dinnerbone — APPROVED** [review]

**Comment by torokati44** [comment]
I took the liberty of resolving a merge conflict manually.

**Pull request merged**